backend/apps/admin/src/problem/problem.service.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException
+} from '@nestjs/common'
+import { isDefined } from 'class-validator'
+import { PrismaService } from '@libs/prisma'
+import { Level } from '@admin/@generated/prisma/level.enum'
+import type { ProblemWhereInput } from '@admin/@generated/problem/problem-where.input'
+import type { TagCreateManyInput } from '@admin/@generated/tag/tag-create-many.input'
+import type { CreateGroupProblemDto } from './dto/create-problem.dto'
+import type {
+  DeleteGroupProblemDto,
+  GetGroupProblemDto,
+  GetGroupProblemsDto
+} from './dto/request-problem.dto'
+import type { UpdateProblem } from './dto/update-problem.dto'
+
+@Injectable()
+export class ProblemService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  private readonly problemSelectOption = {
+    id: true,
+    createdById: true,
+    groupId: true,
+    title: true,
+    description: true,
+    inputDescription: true,
+    outputDescription: true,
+    hint: true,
+    languages: true,
+    timeLimit: true,
+    memoryLimit: true,
+    difficulty: true,
+    source: true,
+    createTime: true,
+    updateTime: true,
+    inputExamples: true,
+    outputExamples: true,
+    problemTestcase: true,
+    problemTag: true
+  }
+
+  async create(problemCreateInput: CreateGroupProblemDto) {
+    const createdBy = {
+      connect: {
+        id: problemCreateInput.createdById
+      }
+    }
+
+    const group = {
+      connect: {
+        id: problemCreateInput.groupId
+      }
+    }
+
+    const problemTestcase = {
+      create: problemCreateInput.problemTestcase
+    }
+
+    let tagsId
+    if (isDefined(problemCreateInput.problemTag)) {
+      const parsedTag = problemCreateInput.problemTag.map(
+        (value): TagCreateManyInput => {
+          return {
+            name: value
+          }
+        }
+      )
+      await this.prisma.tag.createMany({
+        data: parsedTag,
+        skipDuplicates: true
+      })
+      const _tagsId = await this.prisma.tag.findMany({
+        select: { id: true },
+        where: {
+          OR: parsedTag
+        }
+      })
+      tagsId = _tagsId.map((value) => {
+        return {
+          tagId: value.id
+        }
+      })
+    }
+
+    const {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      groupId: _1,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      createdById: _3,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      problemTestcase: _4,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      problemTag: _5,
+      ..._data
+    } = problemCreateInput
```

Comment:
위의 리뷰들을 반영하면 input에서 빼야 할 값은 problemTestcase 하나밖에 남지 않으니까,
```suggestion
    const { testcases, ...problem } = input
    return await this.prisma.problem.create({
      data: {
        problemTestcase: {
          create: testcases
        }
      })
```
이렇게 가면 될 것 같아요!

Reply:
tag의 경우, 위 comment를 반영해 프런트에서 배열로 받아온 것을 객체들의 배열로 전환해 입력되도록 수정했습니다. 따라서, testcase 뿐 아니라 tag도 삭제 시키도록 수정하겠습니다.